`indent` rule false positive

Starting with eslint 2.1.0, I get incorrect `indent` behavior.

With this config:

```json
{
  "rules": {
    "indent": [2, 2]
  }
}
```

And this code:

```js
var info = {
  options: fn(
    true,
    {
      prop: true
    }
  )
}
```

I get this error:

```
/Users/feross/code/standard-stuff/test/t.js
  4:5  error  Expected indentation of 2 space characters but found 4  indent
  5:7  error  Expected indentation of 4 space characters but found 6  indent
  6:5  error  Expected indentation of 2 space characters but found 4  indent

✖ 3 problems (3 errors, 0 warnings)
```

In 2.0.0, this reported no errors. Starting in 2.1.0, it does.

This might be caused by @gyandeeps's fix for https://github.com/eslint/eslint/issues/5237 which was released in 2.1.0.

Same bug I believe... This is causing many of our builds to fail.  Here were my findings:

```javascript
/* eslint indent: [2,2] */
function SillyFunction(arg, arg2) {
  return { irrelevant: true };
}

// validates fine in 2.0 and 2.1
SillyFunction(
  {
    value: true,
  },
  {
    _id: true,
  }
);

var haveFun = function () {

  // No longer validates here in 2.1.0
  SillyFunction(
    {
      value: true,
    },
    {
      _id: true,
    }
  );

  // I guess this is what it wants?  This validates in 2.1.0.
  SillyFunction(
  {
    value: true,
  },
  {
    _id: true,
  }
  );

  return "fun";
};

// Guess not! Same code doesn't validate here, outside of the block.
SillyFunction(
{
  value: true,
},
{
  _id: true,
}
);

haveFun();
```

```bash
$ eslint --no-eslintrc untitled.js 

/private/tmp/untitled.js
  20:5  error  Expected indentation of 2 space characters but found 4  indent
  21:7  error  Expected indentation of 4 space characters but found 6  indent
  22:5  error  Expected indentation of 2 space characters but found 4  indent
  23:5  error  Expected indentation of 2 space characters but found 4  indent
  24:7  error  Expected indentation of 4 space characters but found 6  indent
  25:5  error  Expected indentation of 2 space characters but found 4  indent
  43:1  error  Expected indentation of 2 space characters but found 0  indent
  44:3  error  Expected indentation of 4 space characters but found 2  indent
  45:1  error  Expected indentation of 2 space characters but found 0  indent
  46:1  error  Expected indentation of 2 space characters but found 0  indent
  47:3  error  Expected indentation of 4 space characters but found 2  indent
  48:1  error  Expected indentation of 2 space characters but found 0  indent

✖ 12 problems (12 errors, 0 warnings)
```

--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -431,7 +431,12 @@ module.exports = function(context) {
                 if (parent.type !== "VariableDeclarator" || parentVarNode === parentVarNode.parent.declarations[0]) {
                     if (parentVarNode.loc.start.line === effectiveParent.loc.start.line) {
                         nodeIndent = nodeIndent + (indentSize * options.VariableDeclarator[parentVarNode.parent.kind]);
-                    } else if (parent.type === "ObjectExpression" || parent.type === "ArrayExpression") {
+                    } else if (
+                        parent.type === "ObjectExpression" ||
+                        parent.type === "ArrayExpression" ||
+                        parent.type === "CallExpression" ||
+                        parent.type === "ArrowFunctionExpression"
+                    ) {
                         nodeIndent = nodeIndent + indentSize;
                     }
                 }

--- a/tests/lib/rules/indent.js
+++ b/tests/lib/rules/indent.js
@@ -1053,7 +1053,37 @@ ruleTester.run("indent", rule, {
             "    });\n" +
             "};",
             parserOptions: { ecmaVersion: 6 },
-            options: [4, {"SwitchCase": 1}]
+            options: [4]
+        },
+        {
+            code:
+            "var haveFun = function () {\n" +
+            "    SillyFunction(\n" +
+            "        {\n" +
+            "            value: true,\n" +
+            "        },\n" +
+            "        {\n" +
+            "            _id: true,\n" +
+            "        }\n" +
+            "    );\n" +
+            "};",
+            options: [4]
+        },
+        {
+            code:
+            "let object1 = {\n" +
+            "  doThing() {\n" +
+            "    return _.chain([])\n" +
+            "      .map(v => (\n" +
+            "        {\n" +
+            "          value: true,\n" +
+            "        }\n" +
+            "      ))\n" +
+            "      .value();\n" +
+            "  }\n" +
+            "};",
+            parserOptions: { ecmaVersion: 6 },
+            options: [2]
         }
     ],
     invalid: [

